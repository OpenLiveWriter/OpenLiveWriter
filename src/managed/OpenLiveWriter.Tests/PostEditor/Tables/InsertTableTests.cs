using ApprovalTests;
using ApprovalTests.Reporters;

using NUnit.Framework;

using OpenLiveWriter.PostEditor.Tables;

namespace OpenLiveWriter.Tests.PostEditor.Tables
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class InsertTableTests
    {

        [Test]
        public void Table400PixelsWide3Rows4Columns()
        {
            // Arrange
            var editor = new TestHtmlEditor();
            TableProperties tableProperties = new TableProperties();
            tableProperties.Width = new PixelPercent(400, PixelPercentUnits.Pixels);
            int rows = 3;
            int columns = 4;
            TableCreationParameters tableCreationParamters = new TableCreationParameters(rows, columns, tableProperties);

            // Act
            TableEditor.InsertTable(editor, null, tableCreationParamters);

            // Assert
            Approvals.VerifyHtml(editor.Html);
        }

        [Test]
        public void Table100PercentWide3Rows4Columns()
        {
            // Arrange
            var editor = new TestHtmlEditor();
            TableProperties tableProperties = new TableProperties();
            tableProperties.Width = new PixelPercent(100, PixelPercentUnits.Percentage);
            int rows = 3;
            int columns = 4;
            TableCreationParameters tableCreationParamters = new TableCreationParameters(rows, columns, tableProperties);

            // Act
            TableEditor.InsertTable(editor, null, tableCreationParamters);

            // Assert
            Approvals.VerifyHtml(editor.Html);
        }

        [Test]
        public void TableNoWide3Rows4Columns()
        {
            // Arrange
            var editor = new TestHtmlEditor();
            TableProperties tableProperties = new TableProperties();
            tableProperties.Width = new PixelPercent(0, PixelPercentUnits.Undefined);
            int rows = 3;
            int columns = 4;
            TableCreationParameters tableCreationParamters = new TableCreationParameters(rows, columns, tableProperties);

            // Act
            TableEditor.InsertTable(editor, null, tableCreationParamters);

            // Assert
            Approvals.VerifyHtml(editor.Html);
        }

        [Test]
        public void TableNoWide3Rows4ColumnsCellSpacingPadding()
        {
            // Arrange
            var editor = new TestHtmlEditor();
            TableProperties tableProperties = new TableProperties();
            tableProperties.Width = new PixelPercent(0, PixelPercentUnits.Undefined);
            tableProperties.CellSpacing = "2";
            tableProperties.CellPadding = "0";
            int rows = 3;
            int columns = 4;
            TableCreationParameters tableCreationParamters = new TableCreationParameters(rows, columns, tableProperties);

            // Act
            TableEditor.InsertTable(editor, null, tableCreationParamters);

            // Assert
            Approvals.VerifyHtml(editor.Html);
        }
    }
}
