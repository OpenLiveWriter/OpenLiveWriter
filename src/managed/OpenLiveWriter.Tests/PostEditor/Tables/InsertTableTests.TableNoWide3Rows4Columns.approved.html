<table unselectable="on">
  <tbody>
    <tr>
      <td valign="top"></td>
      <td valign="top"></td>
      <td valign="top"></td>
      <td valign="top"></td>
    </tr>
    <tr>
      <td valign="top"></td>
      <td valign="top"></td>
      <td valign="top"></td>
      <td valign="top"></td>
    </tr>
    <tr>
      <td valign="top"></td>
      <td valign="top"></td>
      <td valign="top"></td>
      <td valign="top"></td>
    </tr>
  </tbody>
</table>